llvm/lib/Passes/StandardInstrumentations.cpp

```diff
@@ -684,9 +686,64 @@ PrintIRInstrumentation::~PrintIRInstrumentation() {
   assert(ModuleDescStack.empty() && "ModuleDescStack is not empty at exit");
 }
 
-void PrintIRInstrumentation::pushModuleDesc(StringRef PassID, Any IR) {
+static SmallString<32> getIRDisplayName(Any IR) {
+
+  auto hashName = [](StringRef name) {
```

Comment:
Hashing creates less readable log names, but serves two purposes:

- Ensures file names are of a legal length
- Without hashing you must truncate the module name etc, which leads to name collisions (ie. two modules with a similar name processed by different threads during thinLTO can have the same log file name)
- We could ensure unique filenames by checking which files already exist, and picking a new random available name, but this can quickly bloat the ir dump directory between successive runs
- Hashing is not guaranteed to be stable between executions, so the output files aren't necessarily the same between runs

It is quite easy to find the file you care about with grep, so I think this is a good compromise between the options

Reply:
> Note hashing is not guaranteed to be stable between executions

I thought hashing strings are stable unless the seed for hashing is changed. If so, can we use a stable hash like `stable_hash_combine_string`? Unless the module name is changed, I'd like to get a stable file name.

Reply:
> I thought hashing strings is stable unless the seed for hashing is changed

I think this is true, and the seed doesn't even seem to change at the moment. That said it looks like the intention is to have it change between executions eventually.

https://github.com/llvm/llvm-project/blob/c60ccfbb898148449946f82cbac6140f1e01cb12/llvm/include/llvm/ADT/Hashing.h#L333-L343

That said, the comments are pretty clear that stability isn't to be relied upon so I think `stable_ash_combine_string` is better for our purposes.

https://github.com/llvm/llvm-project/blob/c60ccfbb898148449946f82cbac6140f1e01cb12/llvm/include/llvm/ADT/Hashing.h#L20-L24